### DISCUSSION TITLE
Documentation

### DISCUSSION THREAD

**Discussion by kenseii** [komoot/photon#535] [Q&A] [answered]
Hello,

Is there any documentation besides the Readme that discusses the way photon indexes data and the way it extracts the name/POI from the data?

Thank you

**Comment by lonvia** [comment, marked as answer]
No, sorry, we only have the source code as documentation. https://github.com/komoot/photon/blob/master/src/main/java/de/komoot/photon/nominatim/NominatimConnector.java is the starting point. This copies the data out of the Nominatim database. https://github.com/komoot/photon/blob/40a2f66406d762156a1186218d67729543072c57/src/main/java/de/komoot/photon/Utils.java#L26 is the central function that converts the whole thing then into ElasticSearch format.

Note, however, that Photon just imports Nominatim data which is already conveniently in a format suitable for geocoding. It's [Nominatim]( https://github.com/osm-search/Nominatim) that does the complex step of processing raw OpenStreetMap data and extracting names and addresses. There is not much documentation on the Nominatim code either, I'm afraid.

**Reply by kenseii** [reply]
Thank you very much for the information,

So when a search request occurs, photon gets the data from elastic indexes, how about when a lat,lon is passed next to the search query is photon using elastic's distance feature query to get the only the results near the given lat long?

Sorry if i might be asking something obvious but i don't use java so navigating the code didn't help much.

**Reply by lonvia** [reply]
It uses a score function to rerank the results. The relevant code is around here: https://github.com/komoot/photon/blob/master/src/main/java/de/komoot/photon/query/PhotonQueryBuilder.java#L109

There is also a bbox parameter that really filters by location, which uses ES's geographic filters: https://github.com/komoot/photon/blob/master/src/main/java/de/komoot/photon/query/PhotonQueryBuilder.java#L127

**Reply by kenseii** [reply]
Thank you very much for the info, it was helpful
